jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageReference.java

```diff
@@ -201,6 +201,16 @@ public boolean usesDefaultTag() {
     return DEFAULT_TAG.equals(tag);
   }
 
+  /**
+   * Gets an {@link ImageReference} with the same registry and repository, but a different tag.
+   *
+   * @param tag the new tag
+   * @return an {@link ImageReference} with the same registry/repository and the new tag
+   */
+  public ImageReference retag(String tag) {
+    return ImageReference.of(registry, repository, tag);
```

Comment:
This is a bit confusing to look at, since `registry`, `repository`, and `tag` are all fields in `ImageReference`, but in this case only `tag` is local. So maybe we should rename `tag` to `newTag` here.